### PR TITLE
Adding the DetailsPageHighlights component

### PR DESCRIPTION
**Pull request by aditirajagopal**
Implemented the Carbon equivalent of https://github.ibm.com/Bluemix/Bluemix.Catalog/blob/master/render-src/components/DetailsPageHighlights.js

To Do: Get feedback, figure out unit tests & scss cleanup

Signed-off-by: Aditi Rajagopal <arajagopal@us.ibm.com>

### CODE REVIEW

**Comment by jrpomeroy** [comment]
@chrisdhanaraj Any way you could take a look at this and provide feedback when you get a minute?  We're hoping to implement this component in our service by end of July so I just want to make sure there is some movement here.  Thanks!

**Comment by chrisdhanaraj** [comment]
Will do, I'll be taking a look through this today or tomorrow

File: components/DetailsPageHighlights/DetailsPageHighlights.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+// import ThumbnailSlider from './ThumbnailSlider'
+// import MarkdownRenderer from './MarkdownRenderer'
```
Comment (hellobrian) [inline components/DetailsPageHighlights/DetailsPageHighlights.js]: Delete any commented code that doesn't help describe or document the code

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+node_modules
+bower_components
+coverage
+public
+reverse-proxy
```
Comment (chrisdhanaraj) [inline .eslintignore]: can you remove these? There is no bower_components, coverage, public, or reverse-proxy in this repository

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "no-script-url": "off"
+  },
+  "settings": {
+    "import/resolver": {
+      "webpack": {
+        "config": "webpack.config.js"
+      }
+    }
+  }
+}
```
Comment (chrisdhanaraj) [inline .eslintrc]: similarly remove these - eslintrc configuration will get standardized across all Carbon repos in a different PR

File: components/DetailsPageHighlights/DetailsPageHighlights.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import MarkdownRenderer from '../MarkdownRenderer/MarkdownRenderer'
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/DetailsPageHighlights.js]: From our other React library, we've been using an `internal` folder for components that aren't really for general consumption. The MarkdownRenderer might belong there instead

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 5]
```diff
@@ -0,0 +1,183 @@
+.bx--artifact-details-highlight-container {
+
+  & .bx--createArtifactFlexGroupColumn {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: why the additional specificity?

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    display: flex;
+    flex-direction: column;
+
+    &.bx--hide {
+      display: none;
+    }
+  }
+
+  & .bx--title-label {
+    font-size: 1.55rem;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Elsewhere we use `rem(pixelvalue)` for font-size. Would be good to standardize around that

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+    font-weight: 200;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Font-weight: 200 is a deprecated weight, swap to `300` for light and  `400` for regular

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+    line-height: 1.5;
+  }
+
+  & .bx--description-label {
+    font-size: rem(14px);
+    font-weight: 300;
+    margin: rem(25px) 0;
+  }
+
+  @include breakpoint(825px) {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: any media query should be embedded in the class that needs it. Additionally (this is new), we're moving away from the `@include breakpoint` to a more standard

`@media (min-width: breakpoint('xs'))`

with the mapping looking like

```
$grid-breakpoints: (sm: 576px, md: 768px, lg: 992px, xl: 1200px, xxl: 1600px);
```

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+    & .bx--createArtifactFlexGroupRow {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: can you double check the specifity requirements for nesting this class?

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+      flex-wrap: wrap;
+    }
+  }
+}
+
+.bx--artifact-details-description-container {
+  margin-bottom: 5px;
+}
+
+@media (max-width: 768px) {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Similar point, all media queries should be nested inside the class

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+    letter-spacing: normal;
+}
+
+.bx--artifact-details-highlight-container {
+  .bx--markdown-container {
+    .underline-text {
+      text-decoration: underline;
+    }
+
+    em {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: not required for the `em` tag

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+      }
+    }
+  }
+}
+
+
+.bx--artifact-details-description,
+.bx--markdown-container {
+  font-size: 0.875em;
+  font-family: "IBM Helvetica", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Not required, the `body` tags will handle the font-family

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+        padding-left: .25rem;
+      }
+    }
+  }
+}
+
+
+.bx--artifact-details-description,
+.bx--markdown-container {
+  font-size: 0.875em;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: We don't use `em` units here

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+  font-family: "IBM Helvetica", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
+}
+
+.bx--artifact-details-description-content {
+  width: 100%;
+  height: auto;
+  overflow: hidden;
+  line-height: 1.2858;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: What does this line-height correspond to

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+
+
+.bx--artifact-details-description,
+.bx--markdown-container {
+  font-size: 0.875em;
+  font-family: "IBM Helvetica", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
+}
+
+.bx--artifact-details-description-content {
+  width: 100%;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Block elements are natively 100% width, unnecessary scss

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+  height: auto;
+  overflow: hidden;
+  line-height: 1.2858;
+
+  .bx--artifact-details-description-bullets {
+    display: flex;
+    justify-content: flex-start;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: flex-start is the initial starting property for `justify-content`, no need to describe it here

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+    flex-direction: column;
+    flex-wrap: wrap;
+
+    width: 100%;
+    font-family: "IBM Helvetica", "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Unnecessary font-family tag

File: components/MarkdownRenderer/markdownrenderer.scss [hunk 18]
```diff
@@ -0,0 +1,74 @@
+.bx--artifact-details-highlight-container {
+    .bx--markdown-container {
+        .underline-text {
```
Comment (chrisdhanaraj) [inline components/MarkdownRenderer/markdownrenderer.scss]: looks like this class is getting reused / redefined a couple times

Comment (chrisdhanaraj) [reply components/MarkdownRenderer/markdownrenderer.scss]: (from the previous scss sheet above), same scss

File: components/MarkdownRenderer/markdownrenderer.scss [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+
+        p {
+            margin-bottom: .625rem;
+
+            &:last-child {
+                margin-bottom: 0;
+            }
+        }
+
+        strong {
```
Comment (chrisdhanaraj) [inline components/MarkdownRenderer/markdownrenderer.scss]: not required

File: package.json [hunk 20]
```diff
@@ -7,13 +7,24 @@
...
+    "markdown-it": "^8.3.1",
+    "markdown-it-sup": "^1.0.0"
+  },
   "devDependencies": {
     "@storybook/addon-centered": "^3.1.2",
     "@storybook/addon-info": "^3.1.6",
     "@storybook/react": "^3.1.6",
     "carbon-components": "^7.12.1",
     "css-loader": "^0.28.4",
+    "babel-eslint": "^6.0.3",
```
Comment (chrisdhanaraj) [inline package.json]: remove the linting dev changes done here

**Review by chrisdhanaraj — CHANGES_REQUESTED** [review]
Same as the other PR - the React itself is mostly fine, but the `scss` needs a lot of cleanup before it can be merged

**Comment by davidmenendez** [comment]
@chrisdhanaraj can you review the latest changes?

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+    letter-spacing: normal;
+}
+
+.bx--artifact-details-highlight-container {
+  .bx--markdown-container {
+    .underline-text {
+      text-decoration: underline;
+    }
+
+    p {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: We don't style elements directly, always through classes. This let's us flex the element type if need be in the future

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 22]
```diff
@@ -0,0 +1,163 @@
...
+    sup {
+      vertical-align: super;
+      font-size: smaller;
+    }
+
+    ul {
+      padding: 1rem;
+      list-style-type: disc;
+
+      ul {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: You're at four levels of specifity here, we cap Carbon at three unless there's a _really_ good reason. I'd look at refactoring and applying the BEM principles to simplify the CSS structure

File: components/DetailsPageHighlights/detailspagehighlights.scss [hunk 23]
```diff
@@ -0,0 +1,163 @@
...
+      flex-direction: row;
+    }
+  }
+
+  .bx--description-bullet-content {
+    width: 100%;
+    padding-left: rem(20px);
+    margin-bottom: rem(24px);
+
+    @media screen and (min-width: 825px) {
```
Comment (chrisdhanaraj) [inline components/DetailsPageHighlights/detailspagehighlights.scss]: Can you see if you can adjust this to one of the breakpoints we already define

File: .storybook/webpack.config.js [hunk 24]
```diff
@@ -1,4 +1,5 @@
 const path = require('path');
+const webpack = require("webpack");
```
Comment (chrisdhanaraj) [inline .storybook/webpack.config.js]: Why is this being added in?

**Review by chrisdhanaraj — CHANGES_REQUESTED** [review]
Apologies for the delay, couple more clean up things here.

**Comment by aditirajagopal** [comment]
@chrisdhanaraj updated based on your comments

**Comment by chrisdhanaraj** [comment]
👍

**Review by chrisdhanaraj — APPROVED** [review]

**Pull request merged**